### PR TITLE
Improve State Management 

### LINKED ISSUE #146: Use Context For State Management
Not sure but just wanted to throw it in: Maybe React's context are a better solution for passing down the citesupport service? See eg https://medium.com/the-guild/injectable-services-in-react-de0136b6d476

_Originally posted by @tobiasdiez in https://github.com/JabRef/JabRef-Word-Addin/pull/103#discussion_r680365611_

### PR DESCRIPTION
**Pull request by mohit038**
Fixes #146 #75 

Added a wrapper to manage the selection state.

### CODE REVIEW

File: src/taskpane/pages/citationStyle.tsx [hunk 1]
```diff
@@ -61,11 +61,8 @@ const classNames = mergeStyleSets({
   ],
 });
 
-interface CitationStyleProps {
-  citeSupport: CiteSupport;
-}
-
-function CitationStyle({ citeSupport }: CitationStyleProps): JSX.Element {
+function CitationStyle(): JSX.Element {
+  const citeSupport = useCiteSupport();
```
Comment (k3KAW8Pnf7mkmdSMPHz27) [inline src/taskpane/pages/citationStyle.tsx:65]: Why is this needed here, does it force re-renders, and if so, why is it needed?

**Review by k3KAW8Pnf7mkmdSMPHz27 — COMMENTED** [review]
Ok, I ended up not having time to finish the review :/
Some initial thoughts/comments are:

* I like the injection and it makes the code look great! 
* There is a lot of code changes :stuck_out_tongue: Perhaps it is needed in this case
* It won't take 9 days until the continuation of the review, from Windows ... :skull: 

LoL, the `office-addin-debugging` just summarized the last 30 min of my life :rofl: 

> Unable to unregister the Office Add-in. Error: Platform not supported: linux.

---

One thing I am not understanding, the "DocumentList"/"DocumentView". Why "Document' instead of citation or something else?

File: src/taskpane/Layout/Layout.tsx [hunk 2]
```diff
@@ -61,8 +57,9 @@ const pivotStyle: Partial<IPivotStyles> = {
   },
 };
 
-function Layout({ citeSupport }: LayoutProps): ReactElement {
+function Layout(): ReactElement {
   const [logoutMutation] = useLogoutMutation();
+  const citeSupport = useCiteSupport();
```
[the remark this answers is not in the document]
Comment (mohit038) [reply src/taskpane/Layout/Layout.tsx:62]: This is a custom hook to get the citeSupport instance for the context. It is not doing any force re-render. We can use ` const citeSupport = useContext(CiteSupportContext);` but I added a wrapper around this, so we don't have to import the actual context. 

``` ruby
export function useCiteSupport(): CiteSupport {
  const context = useContext(CiteSupportContext);
  if (!context) {
    throw new Error(
      "useCiteSupport must be used within a CiteSupportContextProvider. Wrap a parent component in <CiteSupportContextProvider> to fix this error."
    );
  }
  return context;
}
```

[the remark this answers is not in the document]
Comment (k3KAW8Pnf7mkmdSMPHz27) [reply src/taskpane/Layout/Layout.tsx:62]: Oups, sorry my bad. I missed its use on line 113

**Review by k3KAW8Pnf7mkmdSMPHz27 — APPROVED** [review]
This looks good to me!

Not 100% comfortable with all of the React but based on the parts I've tried to read up on it is correct and it works as I expect side-loaded.

I am slightly amused that using **both** Context and reducer appear to have been the right option X)

**My only remaining question** is why the names "DocumentView" and "DocumentList"?

**Review by Siedlerchr — APPROVED** [review]
I can only repeat what Jonatan says

**Comment by mohit038** [comment]
> My only remaining question is why the names "DocumentView" and "DocumentList"?

I don't have any good reason. I thought maybe we should use similar naming here as in jabrefonline. But I guess "Reference" makes more sense here in this scenario, so I've updated the names now. Thanks for your review!

**Review by tobiasdiez — APPROVED** [review]
Sorry for the late review. This looks really nice, good job!
